src/coreclr/inc/gcinfo.h

```diff
@@ -38,6 +38,21 @@ const unsigned   this_OFFSET_FLAG  = 0x2;  // the offset is "this"
 
 #define GCINFO_VERSION 4
 
+#ifdef SOS_INCLUDE
+extern bool IsRuntimeVersionAtLeast(DWORD major);
+// SOS needs to support both v4 and v3 versions.
+//     we can figure which one is used from the major version of the runtime.
+inline int GCInfoVersion()
+{
+    // Since in SOS we only care about ability to parse the GC Info,
+    // we can assume that everything before net10.0 uses GCInfo v3
+    // v2 and v3 had the same format, so for parsing/dumping purposes they are the same.
+    // Also, since runtime cannot parse GC info in nondefault format,
```

Comment:
> since runtime cannot parse GC info in nondefault format,

It is the case at the moment. We had a situation for a while where the runtime parsed multiple version (the older version can come from R2R file). It may be the case again in future.

Reply:
Anything is possible, I guess. I think the main invariant here, which should hold, is that the default GC Info format can be inferred from runtime`s version. It would be highly unusual for the default format to change to change without changing at least minor version of the runtime.

Reply:
Actually, maybe that does not even need mentioning. 

Only the part that we treat v2 and v3 the same needs to be commented as it is a simplification that we can allow because the file format was the same.